### PR TITLE
Fix genesis call ordering

### PR DESCRIPTION
**Pull request by Cyberboss**
This should ensure the genesis call _actually_ runs first.

Credit to @ike709 and @wixoaGit for figuring out initialization order of statics.

### CODE REVIEW

File: code/___genesis_call.dm [hunk 1]
```diff
@@ -39,7 +37,7 @@
  * BYOND loves to tell you about its loving spouse /global
  * But it's actually having a sexy an affair with /static
  * Specifically statics in procs
- * Priority is given to these lines of code in REVERSE order of declaration in the .dme
+ * Priority is given to these lines of code in order of declaration in the .dme
```
Comment (ike709) [inline code/___genesis_call.dm:40]:
```suggestion
 * Priority is given to static var initialization in order of the owning type's type declaration in the .dme
```

**Comment by Boopideedoo** [comment]
So what does this change fix exactly?

**Comment by Cyberboss** [comment]
> So what does this change fix exactly?

Static vars not being profiled

**Comment by Xander3359** [comment]
Wait this has to be first? Was this a recent BYOND change or a new discovery?

**Comment by Cyberboss** [comment]
New discovery

**Comment by ike709** [comment]
> Wait this has to be first? Was this a recent BYOND change or a new discovery?

The static var doesn't necessarily have to be first, but the type declaration for the type that contains the static var does have to be first.

File: code/___genesis_call.dm [hunk 2]
```diff
@@ -25,10 +23,10 @@
  * THE GENESIS CALL
  *
  * THE VERY FIRST LINE OF DM CODE TO EXECUTE
- * Ong this must be done after !!!EVERYTHING!!! else
```
Comment (optimumtact) [inline code/___genesis_call.dm:28]: powerful stuff

**Comment by Cyberboss** [comment]
The plot thickened

**Pull request closed without merging**